### PR TITLE
initialize camera then try socket connection

### PR DESCRIPTION
**Pull request by gferraro**

### CODE REVIEW

File: cmd/leptond/main.go [hunk 1]
```diff
@@ -118,29 +108,26 @@ func runMain() error {
...
 
-		err := runCamera(conf, camera, conn)
-		if err != nil {
-			if _, isNextFrameErr := err.(*nextFrameErr); !isNextFrameErr {
-				return err
+		for {
+			err = runCamera(conf, camera)
+			if err != nil {
+				if isConnectionError(err) {
+					log.Print(err.Error())
```
Comment (mjs) [inline cmd/leptond/main.go]: No need for the call to Error. `Print` and friends will notice that the thing they're printing conforms to the error interface and will do this for you.

File: cmd/leptond/main.go [hunk 2]
```diff
@@ -153,7 +140,36 @@ func runMain() error {
...
+	if err := camera.SetRadiometry(true); err != nil {
+		return nil, err
+	}
+
+	log.Print("opening camera")
+	if err := camera.Open(); err != nil {
+		return nil, err
+	}
+	return camera, nil
+}
```
Comment (mjs) [inline cmd/leptond/main.go:160]: Good idea to extract this. It makes the main func much nicer to read.

**Review by mjs — APPROVED** [review]
Very nice

**Comment by gferraro** [comment]
This was tested on my device, restarted feverscreen while leptond was running, it reconnected almost straight away

File: cmd/leptond/main.go (hunk 1 above)
Comment (gferraro) [reply cmd/leptond/main.go]: oh yea :P

**Pull request merged**